src/drt/src/TritonRoute.cpp

```diff
@@ -86,6 +94,35 @@ void TritonRoute::setDebugPA(bool on)
   debug_->debugPA = on;
 }
 
+void TritonRoute::setDistributed(bool on)
+{
+  distributed_ = on;
+}
+
+void TritonRoute::setWorkerIpPort(const char* in)
+{
+  std::string ip_port = in;
+  size_t pos = ip_port.find("/");
```

Comment:
In most applications I've seen that set the IP and port this way they usually use {IP}:{PORT}, but that makes things difficult for ipv6. The other way I've seen it done is by assuming both a host and a port, and allow the user to override each independently. That would simplify a lot of your logic here.